### PR TITLE
CRUD Operation

### PR DESCRIPTION
**Pull request by degisew**
### To Do list: add & remove
**In this PR I implemented the following features:**

- I removed all hardcoded items from the tasks array and created a new JavaScript file for the new functionality.
- I implemented a function for adding a new task (add a new element to the array).
- I implemented a function for deleting a task (remove an element from the array).
- I implemented a function for editing task descriptions.
- when deleting a task, all remaining items' indexes will be updated.
- All changes to the To-Do List are saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,36 +1,70 @@
...
+// to change the icon when the focus is left.
+allTaskLists.forEach((taskList) => {
+  taskList.addEventListener('focusout', () => {
+    taskList.lastElementChild.previousElementSibling.style.display = 'block';
+    taskList.lastElementChild.style.display = 'none';
+  });
+});
+
+const allTrashBtn = document.querySelectorAll('.trash');
+// console.log(all);
```
Comment (Nemwel-Boniface) [inline src/index.js]: - [ ] Kindly remove all commented out and unused/ unwanted pieces of code from your code base. It makes it look messy and cluttered.

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,70 @@
...
+    e.target.parentElement.remove();
+    Tasks.remove(e.target.id);
+  });
+});
+
+const editTask = document.querySelectorAll('.editable-input-field');
+editTask.forEach((eachInputField, fieldId) => {
+  eachInputField.addEventListener('change', (e) => {
+    // const id = +e.target.id - 1;
+    // console.log(typeof(id));
```
Comment (Nemwel-Boniface) [inline src/index.js]: - [ ] Kindly remove all commented out and unused/ unwanted pieces of code from your code base. It makes it look messy and cluttered.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @degisew            :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job with your Todo list project so far :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] For some reason, I was not able to start your project when I run npm start. Below is the error I was getting:

image


On further inspection, I found out you did not have the "start" scripts, so I added it and it worked perfectly. Below is the correct "scripts" section in your package.json file:

```
"scripts": {
    "start": "webpack serve --open",
    "build": "webpack",
    "dev": "webpack server"
  },
```

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (degisew) [reply src/index.js]: ohh...thank you so much @Nemwel-Boniface. I will fix it.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @degisew, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**